server/src/repositories/patient_repository.ts

```diff
@@ -114,6 +115,44 @@ export class PatientRepository {
         return this.userRepository.buildUsers(res);
     }
 
+    async findPatientsFiltered(testResult: TestResultType, testDateFrom: Date, testDateTo: Date): Promise<User[]> {
+        const client = await this.pool.connect();
+
+        const sql = `
+            SELECT
+                u.user_id,
+                u.email,
+                u.password,
+                u.first_name,
+                u.last_name,
+                u.phone_number,
+                u.gender,
+                u.date_of_birth,
+                u.created_on,
+                u.is_prioritized,
+                r.role_name,
+                a.address_id,
+                a.street_address,
+                a.street_address_line_two,
+                a.city,
+                a.province,
+                a.postal_code,
+                a.country
+            FROM users AS u
+            JOIN patients AS p ON u.user_id = p.patient_id
+            JOIN roles AS r ON u.role_id = r.role_id
+            JOIN addresses AS a ON u.address_id = a.address_id
+            Join test_results As tr On tr.patient_id = p.patient_id
+            WHERE tr.result = $1 AND
+                  tr.testDate >= $2 AND
+                  tr.testDate <= $3
+            ORDER BY u.is_prioritized DESC;
```

Comment:
This will join users on test results. Since a user can have multiple test results you will have duplicate patients being returned. What I think you want to do is to take the last i.e. the most recent test result in the date range and join with that. The idea is a patient is positive if their most recent test result is positive in the date range.

You can do it through SQL but it might be easier to just do it in code, up to you though.